src/semantic_memory/utils.py

```
@@ -0,0 +1,79 @@
+        get_embedding._model = SentenceTransformer(model_name)
+    embedding = get_embedding._model.encode([text])[0]
+    return embedding
+
+_STOP_WORDS = {
+    "the", "and", "or", "a", "an", "of", "to", "in", "for", "with", "on",
+    "by", "at", "from", "is", "are", "was", "were", "be", "been", "being",
+}
+
+def extract_keywords(text: str, max_keywords: int = 5) -> list[str]:
+    """Extract a few salient keywords from the text.
+    Naive implementation: split, lower, filter stop‑words, return most frequent.
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)